### ISSUE TITLE
Item.intersect() fails on CompoundPath

### ISSUE THREAD

**Issue by iconexperience** [paperjs/paper.js#958]
We have not done too much testing on `Item.intersect()`, so I am not really surprised that we still find issues with it.

In this case an intersect between a `CompoundPath` and its bounds results in only part of the `CompoundPath`, while the correct result would cover the whole `CompoundPath`.

![image](https://cloud.githubusercontent.com/assets/2122773/12986691/e9f3f1e8-d0f8-11e5-98ef-1877d7819d5a.png)

```javascript
var p1 = new Path.Rectangle(100, 200, 200, 20);
var p2 = new Path.Rectangle(140, 100, 20, 20);
var cp = new CompoundPath({children: [p1, p2]});

var bounds = new Path.Rectangle(cp.bounds);

var inters = bounds.intersect(cp);

cp.fillColor = "lightgreen";
bounds.strokeColor = "red";
inters.fullySelected = true;
```

<a href="http://sketch.paperjs.org/#S/ddDBCsIwDIDhVyk9KYzhxNPEky8genQeahu3sawtWaeI+O5mdqKCXgol/9dAb9KqFmQudw0EXclEameG+1mR8JlYCQsXsVGhSregg7IlwiSbzRIx/zimy4IK+yTzP2TB4ei+hfajWLvWu96aQU5uuqrRENhc7H2W8LuHezQvdxza7vc27dM4/ia1DUADicM03lkxeJeMTzXi2qEjbguJdVmFkgBsIZ/VyLtAroF3R2DGID6cnnrE6w6QN4DhJFAPS/7iI4FqvOOqk/n+cH8A">Here is the sketch</a>

**Comment by iconexperience** [comment]
Hmm, I first thought this was caused by the new `getWinding()` implementation, but the windings look alright at first sight.

**Comment by iconexperience** [comment]
I have the strong feeling that the cause for this issue is the same as for #923. At the beginning of the boolean operation the bounds rectangle gets subdivided at all intersections with the compound path. As a result all segments of the bottom green rectangle exist twice, and those are the segments that are finally found as being part of the intersection. Finally, when the result is to be created, there are two sets of identical segments for the bottom green rectangle, just like in #923.

So my bold claim is that if we fix #923, we will fix this issue, too.

**Comment by lehni** [comment]
Yes me too, see https://github.com/paperjs/paper.js/issues/923#issuecomment-173937896
